rfc-drafts/stream.md

```diff
@@ -634,4 +649,66 @@ If we introduce `-> Stream` first, we will have to permit `LendingStream` in the
 Additionally, if we introduce `LendingStream` later, we'll have to figure out how
 to convert a `LendingStream` into a `Stream` seamlessly.
 
-Further designing generator functions is out of the scope of this RFC.
+### Differences between Iterator generators and Async generators
```

Comment:
I feel like this whole section helps more the language theorists than the end users among us.

I am pretty much one of the experts in my company for async programming (in Rust and other languages), but have a hard time on figuring out what it wants to convey to me. That there is a difference in pinning is understandable (even though I now 95% of users won't know what pinning really is). But what would I use `PinIterator` for? And why? Maybe leave it out.

I would also encourage again to think about `Stream`s not as a tool for crazy combinators.
The goal of async/await is explicitely to allow people to just write "plain old control flow". If you need combinators - write a for loop. Everyone can do that, and it's in most cases a lot easier for reviewers to figure out whats going on in them than with combinators. If we have any limitations on combinators I still don't think it's a huge loss for practial purposes.

Reply:
There will be plenty of language theorists commenting on the RFC - this is me trying to get ahead of that. And combinators will also most definitely come up in the comments (it certainly has in our pre-RFC discussions). One of the goals of the RFC process is to show that we have throught through as many of the implications as we possibly can. Even though it's not relevant to you personally (or even most of the audience) I would like to leave it in for the few (but vocal!) for whom it is relevant.